src/main/java/com/snowflake/kafka/connector/internal/InternalUtils.java

```diff
@@ -37,6 +38,9 @@
   private static final Logger LOGGER =
     LoggerFactory.getLogger(InternalUtils.class.getName());
 
+  // backoff with 1, 2, 4, 8 seconds
+  public static final int backoffSec[] = {1, 2, 4, 8};
```

Comment:
backoff time interval is set to 1,2,4,8 seconds, which cumulates to 15 seconds. Kafka has a default poll interval of 30 seconds. So our function should not block longer than 30 seconds, otherwise it will trigger rebalance.